### PR TITLE
Add Scanner-API integrationtest

### PR DESCRIPTION
**Pull request by hslusarek**
This PR adds an integration-test which uses all "real" software components of the overall system and simulates the real hardware scanner with a mocked UDP server.

**Note:**
This PR results from the observation that changes to the code base can be made without detecting that the existing functionality was broken.

**Note:**
This PR requires the following PR's to be merged first (rebase this PR than => CI will be green):
- [x] PR #24

### CODE REVIEW

**Comment by martiniil** [comment]
We should merge #18 before this one, there could be conflicts.

File: test/unit_tests/unittest_scanner.cpp [hunk 1]
```diff
@@ -39,50 +37,12 @@ static const std::string DEVICE_IP{ "127.0.0.100" };
...
+TEST(ScannerTest, testGetCompleteScan)
 {
-protected:
-  ScannerConfiguration scanner_config_{ HOST_IP,   HOST_UDP_PORT_DATA, HOST_UDP_PORT_CONTROL,
-                                        DEVICE_IP, START_ANGLE,        END_ANGLE };
-};
-
-typedef ScannerT<psen_scan_v2_test::ScannerControllerMock> MockedScanner;
-
-TEST_F(ScannerTest, testStart)
```
Comment (martiniil) [inline test/unit_tests/unittest_scanner.cpp:53]: Why did you remove these tests?

File: test/integration_tests/integrationtest_scanner_api.cpp [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+#include "psen_scan_v2/scanner_configuration.h"
+#include "psen_scan_v2/scanner.h"
+#include "psen_scan_v2/start_request.h"
+#include "psen_scan_v2/scanner_reply_msg.h"
+
+namespace psen_scan_v2_test
+{
+static const std::string SCANNER_IP_ADDRESS{ "127.0.0.1" };
+static constexpr unsigned short DATA_PORT_OF_SCANNER_DEVICE{ 2000 };
+static constexpr unsigned short CONTROL_PORT_OF_SCANNER_DEVICE{ 3000 };
```
Comment (martiniil) [inline test/integration_tests/integrationtest_scanner_api.cpp:38]: Only a minor thing. I would rename to `SCANNER_UDP_PORT_DATA` and `SCANNER_UDP_PORT_CONTROL` in accordance to the other constants.

File: test/include/psen_scan_v2/async_barrier.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+#ifndef PSEN_SCAN_V2_ASYNC_BARRIER_H
+#define PSEN_SCAN_V2_ASYNC_BARRIER_H
+
+#include <future>
+#include <thread>
+#include <chrono>
+
+namespace psen_scan_v2_test
+{
+class Barrier
```
Comment (martiniil) [inline test/include/psen_scan_v2/async_barrier.h:24]: This class provides the same mechanism as `AsyncTest` from pilz_testutils. Do you think we could replace `AsyncTest` by this class in the future? Now we have both classes used in this package, which I do not like very much.

**Review by martiniil — CHANGES_REQUESTED** [review]
In general :+1:  for the new tests. However I would keep the unittests of the scanner class. Also see the comments below.

File: test/unit_tests/unittest_scanner.cpp (hunk 1 above)
Comment (hslusarek) [reply test/unit_tests/unittest_scanner.cpp:53]: Because they are a duplication with regard to the tests executed in the ScannerAPI integrationtest. -> Done.

File: test/include/psen_scan_v2/async_barrier.h (hunk 3 above)
Comment (hslusarek) [reply test/include/psen_scan_v2/async_barrier.h:24]: I agree, however, for me it is just a first step. If the new Barrier class proves itself, I also suggest to replace the old `AsyncTest` class. For now I would keep it that way. Or what do you suggest?

File: test/integration_tests/integrationtest_scanner_api.cpp (hunk 2 above)
Comment (hslusarek) [reply test/integration_tests/integrationtest_scanner_api.cpp:38]: Done.
